File: game/cosmetic_new/scripts/vscripts/modifier_cosmetic_model.lua
modifier_cosmetic_model_lua = modifier_cosmetic_model_lua or class({})
function modifier_cosmetic_model_lua:IsHidden() return true end
function modifier_cosmetic_model_lua:IsPurgable() return false end
function modifier_cosmetic_model_lua:RemoveOnDeath() return false end
function modifier_cosmetic_model_lua:GetAttributes() return MODIFIER_ATTRIBUTE_PERMANENT end
function modifier_cosmetic_model_lua:GetPriority() return MODIFIER_PRIORITY_LOW end
function modifier_cosmetic_model_lua:DeclareFunctions() return {MODIFIER_PROPERTY_MODEL_CHANGE} end
function modifier_cosmetic_model_lua:OnCreated(kv)
	if not IsServer() then return end
	self.model = kv.model
end
function modifier_cosmetic_model_lua:OnRefresh(kv)
	self:OnCreated(kv)
end
function modifier_cosmetic_model_lua:GetModifierModelChange() return self.model end

File: src/vscripts/cosmetic/modifiers/modifier_cosmetic_model.ts
import { registerModifier } from "../../lib/dota_ts_adapter";
import { BaseModifier } from "../../addon_init";

declare type params = {
	model: string
};

@registerModifier()
export class modifier_cosmetic_model_ts extends BaseModifier {
	model : string = "";

	IsHidden(): boolean {
		return true;
	}

	IsPurgable(): boolean {
		return false;
	}

	RemoveOnDeath(): boolean {
		return false;
	}

	GetAttributes(): ModifierAttribute {
		return ModifierAttribute.PERMANENT;
	}

	GetPriority(): ModifierPriority {
		return ModifierPriority.LOW;
	}

	DeclareFunctions(): ModifierFunction[] {
		return [ModifierFunction.MODEL_CHANGE];
	}

	OnCreated(kv: params): void {
		if (!IsServer()) {
			return;
		}

		this.model = kv.model;
	}

	OnRefresh(kv: params): void {
		this.OnCreated(kv);
	}

	GetModifierModelChange(): string {
		return this.model;
	}
}